crates/re_viewer/src/app.rs

```diff
@@ -304,25 +312,20 @@ impl App {
                 store_hub.remove_recording_id(&recording_id);
             }
 
-            #[cfg(not(target_arch = "wasm32"))]
-            SystemCommand::LoadRrdPath(path) => {
-                let with_notification = true;
-                if let Some(rrd) = crate::loading::load_file_path(&path, with_notification) {
-                    let store_id = rrd.store_dbs().next().map(|db| db.store_id().clone());
-                    store_hub.add_bundle(rrd);
-                    if let Some(store_id) = store_id {
-                        store_hub.set_recording_id(store_id);
-                    }
-                }
-            }
+            SystemCommand::LoadDataSource(data_source) => {
+                let egui_ctx = self.re_ui.egui_ctx.clone();
+                let wake_up_ui_on_msg = Box::new(move || {
+                    // Spend a few more milliseconds decoding incoming messages,
+                    // then trigger a repaint (https://github.com/rerun-io/rerun/issues/963):
+                    egui_ctx.request_repaint_after(std::time::Duration::from_millis(10));
```

Comment:
I'm probably missing something here but isn't that redundant with the waking up performed by `wake_up_ui_thread_on_each_msg` in `add_receiver()`?

Reply:
`wake_up_ui_thread_on_each_msg` spawns a thread to listen for new messages and then wake the ui, and because of that it doesn't work on web (where there are no threads). So we need the above for web.

Reply:
I added a clarifying comment

Reply:
👍🏻